code/game/objects/structures/tables_racks.dm

```diff
@@ -356,8 +360,9 @@
 		var/obj/item/weapon/grab/G = W
 		if (istype(G.affecting, /mob/living))
 			var/mob/living/M = G.affecting
-			if (G.state < 2)
+			if (G.state < GRAB_AGGRESSIVE)
 				if(user.a_intent == I_HURT)
+					G.affecting.loc = src.loc
```

Comment:
This should be `G.affecting.forceMove(loc)`, so it properly updates lighting and other things.